core/src/main/java/hudson/model/Queue.java

```diff
@@ -1460,54 +1460,64 @@ public void maintain() {
             // allocate buildable jobs to executors
             for (BuildableItem p : new ArrayList<BuildableItem>(
                     buildables)) {// copy as we'll mutate the list in the loop
-                // one last check to make sure this build is not blocked.
-                if (isBuildBlocked(p)) {
+                if (p.task instanceof FlyweightTask) {
```

Comment:
With this change the FlyweightTask cannot be blocked by another buildable job that has just started running...

JENKINS-28926 was related to `Task.isBuildBlocked()` being determined by the other executing jobs. it is only safe to assume that `isBuildBlocked(p)` for the very first `p` in `buildables`... once we have updated the snapshot (by starting another task from `buildables` building all the remaining tasks could *potentially* have their `isBuildBlocked()` return `true`.

Smells like a regression